packages/maskbook/src/plugins/Ideamarket/define.tsx

```diff
@@ -0,0 +1,58 @@
+import { PluginConfig, PluginStage, PluginScope } from '../types'
+import { usePostInfoDetails } from '../../components/DataSource/usePostInfo'
+import { getAllListings } from './api'
+import type { getAllListingsData } from './types'
+import LogoButton from './UI/LogoButton'
+import { isArray } from 'lodash-es'
+import Logo from '../../extension/options-page/DashboardComponents/MaskbookLogo'
+
+let ideaTokens: getAllListingsData[]
+
+const fetchAll = async () => {
+    const result = await getAllListings()
```

Comment:
What if the dataset is very large?

Reply:
Then we'd probably be better off with my previous solution..